### PR TITLE
add tests to refactor how multiple errors per liine are handled

### PR DESCRIPTION
**Pull request by paulortman**

### CODE REVIEW

File: tests/test_yaml.py [hunk 1]
```diff
@@ -131,3 +107,79 @@ def test_empty_yaml():
...
+    - one
+    - two
+    - three"""
+
+    config = ProjectLoader(text)
+    assert not config.is_valid()
+
+    expected_error = """\
+deadbeef: deadbeef
+# ^
```
Comment (davegaeddert) [inline tests/test_yaml.py]: I suppose there's not much we can do to make these line up and still have a valid yaml? Unless we repeated the line above in a comment also?
```yaml
deadbeef: deadbeef
# deadbeef: deadbeef
# ^
# ---------
```

Comment (paulortman) [inline tests/test_yaml.py]: We could certainly do that, or else indent every line by 2, except our comments.  Whichever you want.

Comment (davegaeddert) [inline tests/test_yaml.py]: Maybe try the commented repeat line first? Indenting everything could probably work, seems a little more risky though (2 spaces vs 4 spaces, etc).

Comment (paulortman) [inline tests/test_yaml.py]: @davegaeddert Okay, done.

**Pull request merged**